rcl/include/rcl/publisher.h

```diff
@@ -20,7 +20,7 @@ extern "C"
 {
 #endif
 
-#include "rosidl_generator_c/message_type_support.h"
+#include "rosidl_generator_c/message_type_support_struct.h"
```

Comment:
Got it. I'll revert it. I think I was trying to fix issues that were related to the linking problems we had earlier.